data_collection/gazette/spiders/base/municipioonline.py

```diff
@@ -0,0 +1,48 @@
+import re
+
+import scrapy
+from dateutil.rrule import DAILY, rrule
+
+from gazette.items import Gazette
+from gazette.spiders.base import BaseGazetteSpider
+
+
+class BaseMunicipioOnlineSpider(BaseGazetteSpider):
+    allowed_domains = ["municipioonline.com.br"]
+
+    def start_requests(self):
+        url = f"https://www.municipioonline.com.br/{self.url_uf}/prefeitura/{self.url_city}/cidadao/diariooficial"
+        yield scrapy.Request(url)
+
+    def parse(self, response):
+        for dt in rrule(freq=DAILY, dtstart=self.start_date, until=self.end_date):
+            formdata = {
+                "__EVENTTARGET": "ctl00$body$txtDataEdicao",
+                "ctl00$body$txtDataEdicao": dt.strftime("%d/%m/%Y"),
+            }
+
+            yield scrapy.FormRequest.from_response(
+                response,
+                formdata=formdata,
+                callback=self.parse_page,
+                cb_kwargs={"page_date": dt.date()},
+            )
+
+    def parse_page(self, response, page_date):
+        editions_list = response.xpath("//*[@id='body_ddlEdicao']/option").getall()
+
+        if "inexistente" not in editions_list[0]:
+            for raw_edition in editions_list:
+                edition_number = re.search(r">(\w*)/", raw_edition).group(1)
+                url_path = re.search(r'"(.*).pdf"', raw_edition).group(1)
+                gazette_url = response.urljoin(
+                    f"diariooficial/diario?n=diario.pdf&l={url_path}"
+                )
```

Comment:
Aqui vejo que o regex pode ser substituído por outras alternativas mais legíveis se a gente fizer uso dos elementos das edições e não dos textos (tirando o `.getall()`).

1. `editions_list` passa a ser uma lista de elementos, aí tu pode usar o método `.re` pra buscar por `inexistente` e não precisaria mais fixar o elemento 0 da lista (e se quiser a mesma funcionalidade de antes é só usar `.re_first`)
2. `edition_number` é o texto do elemento e dá pra usar seletor normalmente
3. `url_path` é um nome de arquivo com extensão, assim tu pode usar a biblioteca `pathlib` pra pegar só o nome do arquivo sem a extensão

```suggestion
        editions_list = response.css("#body_ddlEdicao option")

        if editions_list.re(r"inexistente", re.IGNORECASE):
            return

        for raw_edition in editions_list:
            edition_number = raw_edition.xpath("./text()").get()
            url_path = Path(raw_edition.attrib["value"]).stem
            gazette_url = response.urljoin(
                f"diariooficial/diario?n=diario.pdf&l={url_path}"
            )

```

Reply:
Aqui seria necessário um `from pathlib import Path`